### PR TITLE
Form 21-8940: Ability to populate Impact on recent job applications questions #14538

### PR DESCRIPTION
**Pull request by zurbergram**
## Description

As a Veteran submitting a claim for Individual Unemployability on the Vets.gov platform, I need the ability to populate the recent job applications, so that I can provide additional evidence associated to my Individual Unemployability while completing the Form 21-8940 application interview.

## Testing done
- [x] manually tested
- [x] Unit tests

## Screenshots
screen shot 2018-12-04 at 2 50 51 pm
screen shot 2018-12-04 at 2 51 02 pm

## Acceptance criteria
- [x] The user is able to populate the following: 
- [x] Have you tried to find a steady job since you became too disabled to work? { radio }
Yes
No
- [x] { If yes }
- [x] Employer’s name { text field }
- [x] Address {address pattern}
- [x] Type of work { text field }
- [x] Date you applied { date picker }
- [x] The user is able to select the help text
- [x] The user is able to add another
- [x] The user is able to move to the next step in the process, Story#14539: Form 21-8940: Ability to populate Past Education and Training questions
- [x] Field validations (if applicable per the linked document for Validations above) are enforced in the UI
- [x] Error messages (if applicable per the linked document for Validations above) are displayed in the UI

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/disability-benefits/all-claims/pages/recentJobApplications.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+          'ui:validations': [validateZIP],
+        },
+      }),
+      workType: {
+        'ui:title': 'Type of work',
+      },
+      date: currentOrPastDateUI('Date you applied'),
+    },
+  },
+  substantiallyGainfulEmploymentInfo: {
```
Comment (aaronjlech) [inline src/applications/disability-benefits/all-claims/pages/recentJobApplications.js]: can we change this to `'view: substantiallyGainfulEmploymentInfo'`. Or something with `view:` in it. We use that to symbolize UI related data fields/content fields(things not in the form Schema).

File: src/applications/disability-benefits/all-claims/pages/recentJobApplications.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+} from '../content/recentJobApplications';
+
+import {
+  uiSchema as addressUI,
+  schema as addressSchema,
+} from '../../../../platform/forms/definitions/address';
+
+const address = addressSchema(fullSchema);
+
+import { validateZIP } from '../validations';
```
Comment (aaronjlech) [inline src/applications/disability-benefits/all-claims/pages/recentJobApplications.js]: Should we move all of the imports above the `const` ? That way they are all relatively in the same location?

**Review by aaronjlech — CHANGES_REQUESTED** [review]
Looks good for the most part. Just 2 little semantic/organization things.

**Review by aaronjlech — APPROVED** [review]
Looks good!

**Comment by zurbergram** [comment]
@department-of-veterans-affairs/code-reviewers-for-external-teams-group - Could I get a code review on this at your earliest convenience? Thanks!

**Review by rianfowler — APPROVED** [review]
LGTM

**Pull request merged**